sagemaker-clarify/fairness_and_explainability/explainability_with_pdp.ipynb

```diff
@@ -0,0 +1,748 @@
+   "outputs": [],
+   "source": [
+    "pdp_explainability_output_path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f3dbfd29-f68f-4194-a66b-e31fa29267b6",
+   "metadata": {},
+   "source": [
+    "For example, you can also download a copy of the html report and view it in-place here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c682916b-d489-4d4c-83b6-9d17a149ef20",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!aws s3 cp {pdp_explainability_output_path}/report.html ./explainability-report.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f9ba7764-8c1b-4314-bbb8-3d2912430c73",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import IPython\n",
+    "\n",
+    "IPython.display.HTML(filename=\"explainability-report.html\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "950bd78f-eb06-4103-9e6f-9f0394fa2398",
+   "metadata": {},
+   "source": [
+    "### PDP Analysis with SHAP"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4ba3937-a7fb-4765-84e5-5e55392d3320",
+   "metadata": {},
+   "source": [
+    "Instead of specifying the features we want to run PDP analysis on, we can use [SHAP values](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-shapley-values.html) to rank feature importance and select the `top_k_features` to plot PDPs. SageMaker Clarify uses Kernel SHAP to explain the amount of contribution that each input feature makes to the final decision. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8cde1f64-386c-4828-8529-05fa0afb58ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pdp_config = clarify.PDPConfig(\n",
+    "    top_k_features=5,\n",
+    "    grid_resolution=25,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab4392c2-c760-4f62-b317-7c4541a3b84d",
+   "metadata": {},
+   "source": [
+    "The Kernel SHAP algorithm requires a [baseline for explainability](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-feature-attribute-shap-baselines.html), also known as a background dataset. If a `baseline` is not provided, it will be automatically computed as the cluster centers of the dataset by SageMaker Clarify using K-means or K-prototypes. Baseline dataset type shall be the same as `dataset_type` of `DataConfig`, and baseline samples shall only include features. By definition, `baseline` should either be a S3 URI to the baseline dataset file, or an in-place list of samples. In this case we chose the latter, and put the first sample of the test dataset to the list."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "064119ba-6974-4744-9237-dbd14238de6a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "shap_config = clarify.SHAPConfig(\n",
+    "    baseline=[test_features.iloc[0].values.tolist()],\n",
+    "    num_samples=15,\n",
```

Comment:
Let us remove `num_samples` from here and let the constructor use the default value. If not, we should explain what this parameter does.